### PR TITLE
Install build artifacts into shared folder

### PR DESCRIPTION
**Pull request by iusyk**
DomD artifacts, used by Dom0 to construct the final wic image, are:
1. Boot files: xen, device trees, HDMI firmware etc.
2. DomD root file system.

Install these into shared folder, so Dom0 can use the artifacts
to construct the final wic image.

Signed-off-by: Usyk Ihor <ihor_usyk@epam.com>

### CODE REVIEW

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend [hunk 1]
```diff
@@ -29,17 +31,57 @@ python __anonymous () {
...
+do_collect_artifacts() {
+    # Generate u-boot image for initramfs
+     ${STAGING_BINDIR_NATIVE}/uboot-mkimage -A arm64 -O linux -T ramdisk -C gzip -n "uInitramfs" \
+         -d ${DEPLOYDIR}-image-complete/${IMAGE_LINK_NAME}.cpio.gz ${DEPLOY_DIR_IMAGE}/${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.cpio.gz.uInitramfs
+     ln -sfr  ${DEPLOY_DIR_IMAGE}/${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.cpio.gz.uInitramfs ${DEPLOY_DIR_IMAGE}/uInitramfs
+    # Collect all the boot files
+    rm -rf ${DOM0_BOOT_FILES_DIR}
+    install -d ${DOM0_BOOT_FILES_DIR}
+
+    # Install bootloader files built by DomD into Dom0 deploy dir,
```
Comment (andr2000) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend]: Renesas doesn't have bootloader files, but rather boot files

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend [hunk 2]
```diff
@@ -29,17 +31,57 @@ python __anonymous () {
...
+    # so all the artifacts required for flashing are at the same place
+
+    find ${XT_DIR_ABS_SHARED_BOOT_DOMD}/* -type f | xargs -I{} cp -f {} ${DEPLOY_DIR_IMAGE}
+
+    # Copy all the boot files generated by DomD...
+    if [ -f ${XT_DIR_ABS_SHARED_BOOT_DOMD}/dom0.dtb ]; then
+        cp -fL ${XT_DIR_ABS_SHARED_BOOT_DOMD}/dom0*.dtb ${DOM0_BOOT_FILES_DIR}
```
Comment (andr2000) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend]: why do you need to copy all by mask? I think dom0.dtb is enough
IMO, we just need to follow the symlink while copying

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend [hunk 3]
```diff
@@ -29,17 +31,57 @@ python __anonymous () {
...
+    fi
+    # copy xen and xen policy
+     if [ -d ${XT_DIR_ABS_SHARED_BOOT_DOMD}/xen ]; then
+        cp -fL ${XT_DIR_ABS_SHARED_BOOT_DOMD}/xen/* ${DOM0_BOOT_FILES_DIR}
```
Comment (andr2000) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend]: ditto

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend [hunk 4]
```diff
@@ -29,17 +31,57 @@ python __anonymous () {
...
+        ls ${DOM0_BOOT_FILES_DIR}/xen-*.uImage | head -1 | xargs -I{} ln -sfr {} ${DOM0_BOOT_FILES_DIR}/xen
+        ls ${DOM0_BOOT_FILES_DIR}/xenpolicy-* | head -1 | xargs -I{} ln -sfr {} ${DOM0_BOOT_FILES_DIR}/xenpolicy
+    fi
+    # but use the kernel from Dom0
+    ln -sfr ${DEPLOY_DIR_IMAGE}/Image ${DOM0_BOOT_FILES_DIR}
+    # and use initramfs image of Dom0
+    ln -sfr ${DEPLOY_DIR_IMAGE}/uInitramfs ${DOM0_BOOT_FILES_DIR}
+
+    # Make DomD root file system visible to wic, put it under DEPLOY_DIR_IMAGE
+    ls ${XT_DIR_ABS_SHARED_BOOT_DOMD}/*.ext4 | head -1 | xargs -I{} cp -f {} ${DEPLOY_DIR_IMAGE}/domd.ext4
```
Comment (andr2000) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend]: what about domu?

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/wic/prod-devel.wks [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# short-description: Create SD card image with a boot partition
+# long-description:
+# Create an image that can be written onto a SD card using dd for use
+# with i.MX SoC family
+# It uses u-boot + other binaries gathered together on imx-boot file
+#
+#part u-boot --source rawcopy --sourceparams="file=imx-boot" --ondisk mmcblk --no-table --align 32
```
Comment (andr2000) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/wic/prod-devel.wks]: pls remove

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/wic/prod-devel.wks [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+part / --source rootfs --rootfs-dir=rootfs_boot --ondisk mmcblk --fstype=vfat --label kernel --active --align 16384 --size 32
+part --source rawcopy --sourceparams=file="domd.ext4" --ondisk mmcblk --fstype=ext4 --label domd --align 8192 --size 2G
+
+#bootloader --ptable msdos
```
Comment (andr2000) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/wic/prod-devel.wks]: I think this needs to be uncommented. Did you try to mount the resulting wic?

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend [hunk 7]
```diff
@@ -29,17 +31,57 @@ python __anonymous () {
...
 }
 
-generate_uboot_image() {
-    ${STAGING_BINDIR_NATIVE}/uboot-mkimage -A arm64 -O linux -T ramdisk -C gzip -n "uInitramfs" \
-        -d ${DEPLOYDIR}-image-complete/${IMAGE_LINK_NAME}.cpio.gz ${DEPLOY_DIR_IMAGE}/${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.cpio.gz.uInitramfs
-    ln -sfr  ${DEPLOY_DIR_IMAGE}/${IMAGE_NAME}${IMAGE_NAME_SUFFIX}.cpio.gz.uInitramfs ${DEPLOY_DIR_IMAGE}/uInitramfs
+IMAGE_ROOTFS_SIZE = "65535"
+
+WKS_FILE = "prod-devel.wks"
+IMAGE_FSTYPES += "wic.xz wic.bmap"
```
Comment (andr2000) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend]: please check that bmap tool supports compressed images

File: recipes-domu/domu-image-weston/files/meta-xt-prod-extra/recipes-graphics/images/core-image-weston.bbappend [hunk 8]
```diff
@@ -17,10 +19,18 @@ IMAGE_INSTALL_remove = " \
...
 populate_vmlinux () {
     find ${STAGING_KERNEL_BUILDDIR} -iname "vmlinux*" -exec mv {} ${DEPLOY_DIR_IMAGE} \; || true
 }
 
+addtask populate_artifacts after do_image_complete before do_build
+do_populate_artifacts() {
+    ln -sfr ${DEPLOY_DIR_IMAGE}/${IMAGE_LINK_NAME}.ext4 ${XT_DIR_ABS_SHARED_BOOT_DOMU}
+}
+
+
```
Comment (andr2000) [inline recipes-domu/domu-image-weston/files/meta-xt-prod-extra/recipes-graphics/images/core-image-weston.bbappend]: extra new line

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/wic/prod-devel.wks (hunk 6 above)
Comment (iusyk) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/wic/prod-devel.wks]: here we had a word that it i not necessary , there were an error (i do not remember what was exactly) , but i can uncomment and check the result

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend (hunk 2 above)
Comment (iusyk) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend]: I am not sure if name will be permanent , but  if yes - i do

Comment (iusyk) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend]: fixed

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend (hunk 4 above)
Comment (iusyk) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend]: fixed

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/wic/prod-devel.wks (hunk 5 above)
Comment (iusyk) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/wic/prod-devel.wks]: fixed

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend (hunk 7 above)
Comment (iusyk) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend]: > please check that bmap tool supports compressed images

@andr2000 my understanding- yes
"
_Raw images compress extremely well because the holes are essentially zeroes,
which compress perfectly. This is why 3.7GiB Tizen IVI raw images, which
contain about 1.1GiB of mapped blocks, take only 300MiB in a compressed form.
And the important point is that you  need to decompress them only while
flashing. The bmaptool does this "on-the-fly"._
"

from https://github.com/intel/bmap-tools/blob/master/docs/README

File: recipes-domu/domu-image-weston/files/meta-xt-prod-extra/recipes-graphics/images/core-image-weston.bbappend (hunk 8 above)
Comment (iusyk) [reply recipes-domu/domu-image-weston/files/meta-xt-prod-extra/recipes-graphics/images/core-image-weston.bbappend]: fixed

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend (hunk 7 above)
Comment (iusyk) [reply recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend:37]: http://manpages.ubuntu.com/manpages/xenial/man1/bmaptool.1.html
IMAGE may be compressed,  in  which  case  bmaptool  decompresses  it  on-the-fly.   The
         compression  type  is  detected  by  the file extension and the following extensions are
         supported:

             1. ".gz", ".gzip", ".tar.gz" and ".tgz" for files and tar archives  compressed  with
             "gzip" program
             2.  ".bz2",  "tar.bz2",  ".tbz2",  ".tbz",  and  ".tb2"  for  files and tar archives
             compressed with "bzip2" program
             3. ".xz", ".tar.xz", ".txz" for files and tar archives compressed with "xz" program
             4. ".lzo", "tar.lzo", ".tzo" for  files  and  tar  archives  compressed  with  "lzo"
             program

File: recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend [hunk 9]
```diff
@@ -29,17 +31,57 @@ python __anonymous () {
...
 
 populate_vmlinux () {
     find ${STAGING_KERNEL_BUILDDIR} -iname "vmlinux*" -exec mv {} ${DEPLOY_DIR_IMAGE} \; || true
 }
 
-IMAGE_POSTPROCESS_COMMAND += " generate_uboot_image; populate_vmlinux; "
-
-IMAGE_ROOTFS_SIZE = "65535"
-
+IMAGE_POSTPROCESS_COMMAND += " populate_vmlinux; "
```
Comment (andr2000) [inline recipes-dom0/dom0-image-thin-initramfs/files/meta-xt-prod-extra/recipes-core/images/core-image-thin-initramfs.bbappend:87]: How is this change related?

File: recipes-domu/domu-image-weston/files/meta-xt-prod-extra/recipes-graphics/images/core-image-weston.bbappend [hunk 10]
```diff
@@ -17,10 +19,18 @@ IMAGE_INSTALL_remove = " \
...
 
 populate_vmlinux () {
     find ${STAGING_KERNEL_BUILDDIR} -iname "vmlinux*" -exec mv {} ${DEPLOY_DIR_IMAGE} \; || true
 }
 
+addtask populate_artifacts after do_image_complete before do_build
+do_populate_artifacts() {
+    ln -sfr ${DEPLOY_DIR_IMAGE}/${IMAGE_LINK_NAME}.ext4 ${XT_DIR_ABS_SHARED_BOOT_DOMU}
+}
+
```
Comment (andr2000) [inline recipes-domu/domu-image-weston/files/meta-xt-prod-extra/recipes-graphics/images/core-image-weston.bbappend:34]: extra lines

**Comment by iusyk** [comment]
not relevant anymore, due to the usage of 'moulin'
